cif2cell/uctools.py

```diff
@@ -1209,10 +1210,13 @@ def getFromCIF(self, cifblock=None):
                     else:
                         raise SymmetryError(
                             "CIF file contains neither space group symbols nor space group number.")
-            try:
-                self.HallSymbol = HM2Hall[self.HMSymbol]
-            except:
-                self.HallSymbol = "Unknown"
+            else:
+                try:
+                    self.HallSymbol = HM2Hall[self.HMSymbol]
+                except:
+#                    self.HallSymbol = "Unknown"
+                    raise SymmetryError(
```

Comment:
If I understand correctly, this is a possibly breaking change (CIF files that would pass before this change no longer pass).

I would suggest to reserve such changes for major version bumps, unless there are strong reasons to "fix" this

Reply:
As far as I understood, if self.HallSymbol becomes "Unknown" the structure cannot be defined and there will be error in other parts of the code, for example in line 1235 of uctools.py:
```
         if type(eqsites) == type(None):
             eqsites = SymOpsHall[self.HallSymbol]

```
since SymOpsHall['Unknown'] is not a valid key (what happens in #34 ). But we can leave it as it was before, I thought raising an error was nicer.
The other change that I made in that portion of the code (added an "else:" in line 1214) though I think it's important: in cases where the space group number was defined, but not the H-M, the Hall symbol was correctly given in line 1199, but then was overwritten to "Unknown" in line 1217 (again, this is what happens in #34 ).

Hope this was clear.

Reply:
Thanks for the explanation- I didn't realize this leads to errors in the code elsewhere. Under these circumstances I agree raising an exception is better.
And yes, I saw the other change you made and agree it also fixes a bug